agent/engine/task_manager_test.go

```diff
@@ -15,19 +15,436 @@
 package engine
 
 import (
+	"sync"
 	"testing"
 	"time"
 
 	"github.com/aws/amazon-ecs-agent/agent/api"
 	"github.com/aws/amazon-ecs-agent/agent/engine/dockerstate/mocks"
 	"github.com/aws/amazon-ecs-agent/agent/engine/testdata"
+	"github.com/aws/amazon-ecs-agent/agent/eventstream"
 	"github.com/aws/amazon-ecs-agent/agent/statemanager"
 	"github.com/aws/amazon-ecs-agent/agent/utils/ttime/mocks"
 	"github.com/stretchr/testify/assert"
 
 	"github.com/golang/mock/gomock"
+	"golang.org/x/net/context"
 )
 
+func TestContainerNextStateFromNone(t *testing.T) {
```

Comment:
To clarify: the advantage of doing this is that each subtest will have independent output so you can see which test case went wrong.

Here's a simple one I just did:

```go
var tests = []struct {
	a int
	b bool
}{
	{1, false},
	{2, true},
	{3, true},
}

func TestSubTest(t *testing.T) {
	for _, subtest := range tests {
		t.Run(fmt.Sprintf("%d: %v", subtest.a, subtest.b), func(t *testing.T) {
			if (subtest.a%2 == 0) != subtest.b {
				t.Error("Unexpected bool!")
			}
		})
	}
}
```

```
--- FAIL: TestSubTest (0.00s)
    --- FAIL: TestSubTest/3:_true (0.00s)
    	my_test.go:25: Unexpected bool!
```